File: gluecodium/src/test/resources/smoke/structs/output/android/com/example/smoke/StructsWithMethods.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
public final class StructsWithMethods {
    public static final class Vector {
        public double x;
        public double y;
        public Vector(final double x, final double y) {
            Vector _other = create(x, y);
            this.x = _other.x;
            this.y = _other.y;
        }
        public Vector(@NonNull final StructsWithMethods.Vector other) throws ValidationUtils.ValidationException {
            Vector _other = create(other);
            this.x = _other.x;
            this.y = _other.y;
        }
        public native double distanceTo(@NonNull final StructsWithMethods.Vector other);
        @NonNull
        public native StructsWithMethods.Vector add(@NonNull final StructsWithMethods.Vector other);
        public static native boolean validate(final double x, final double y);
        private static native Vector create(final double x, final double y);
        private static native Vector create(@NonNull final StructsWithMethods.Vector other) throws ValidationUtils.ValidationException;
    }
}

File: gluecodium/src/test/resources/smoke/structs/output/android-kotlin/com/example/smoke/StructsWithMethods.kt
/*

 *
 */

package com.example.smoke


class StructsWithMethods {

    class Vector {
        var x: Double
        var y: Double


        constructor(x: Double, y: Double) {
            val _other = create(x, y)
            this.x = _other.x
            this.y = _other.y
        }
    @Throws (ValidationUtils.ValidationException::class)
        constructor(other: StructsWithMethods.Vector) {
            val _other = create(other)
            this.x = _other.x
            this.y = _other.y
        }




        external fun distanceTo(other: StructsWithMethods.Vector) : Double
        external fun add(other: StructsWithMethods.Vector) : StructsWithMethods.Vector


        companion object {
            @JvmStatic external fun validate(x: Double, y: Double) : Boolean
            @JvmStatic external fun create(x: Double, y: Double) : Vector
            @Throws (ValidationUtils.ValidationException::class) @JvmStatic external fun create(other: StructsWithMethods.Vector) : Vector
        }
    }








}

